Source/Dafny/Reporting.cs

```diff
@@ -235,4 +247,19 @@ public class ErrorReporterWrapper : BatchErrorReporter {
       return WrappedReporter.Message(source, level, tok, msgPrefix + msg);
     }
   }
+
+  public class SimpleErrorCorrector : ErrorReporter {
+    public List<string> CollectedMessages = new();
```

Comment:
I had considered something like `List<Tuple<string,IToken>>` instead of two lists, but not dictionaries, since they don't represent key/values. For this class, I only need to recover the last message and token, so I could simplify it further if you wish.